bundles/org.openhab.binding.magentatv/README.md

```diff
@@ -0,0 +1,257 @@
+# MagentaTV Binding
+
+This Binding allows controlling the Deutsche Telekom Magenta TV Media Receiver series MR4xx and MR2xx (Telekom NGTV / Huawei Envision platform). 
+The Binding does NOT support MR3xx/1xx (old Entertain system based on Microsoft technology)!
+
+Media Receivers are automatically discovered.
+You could send keys as you press them on the remote and the binding receives program information when the channel is switched (no
+The binding provides device discovery, sending keys for the remote and also receiving program information/events.
+
+## Supported Things
+
+|Thing    |Type                                                                    |
+|---------|----------------------------------------------------------------------- |
+|receiver |A MagentaTV Receiver, the binding supports multiple models (see above). |
+
+### Supported Models
+
+* Deutsche Telekom Media Receiver MR401B - fully supported
+* Deutsche Telekom Media Receiver MR201  - fully supported
+* Deutsche Telekom Media Receiver MR400  - supported with minor restrictions (POWER key)
+* Deutsche Telekom Media Receiver MR200  - supported with minor restrictions (POWER key)
+* Deutsche Telekom Media Receiver MR601  - should be supported (not verified)
+* Deutsche Telekom Media Receiver MR3xx  - NOT supported (different platform)
+* Deutsche Telekom Media Receiver MR1xx  - NOT supported (different platform)
+
+## Auto Discovery
+
+UPnP will be used to discover receivers on the local network and discover the nessesary parameters.
+The receiver needs to be powered on to get discovered.
+
+Once the receiver is discovered it can be added from the Inbox.
+Make sure to set the parameter `userId` in the Thing configuration after adding the new thing, see section Thing Configuration.
+
+Note:
+The binding uses the network settings in openHAB's system configuration to determine the local IP address.
+The device can't be discovered if the openHAB system and receiver are not on the same network (IP/Netmask).
+In this case you need to add the Thing manually or use textual configuration (.things).
+
+If you are running openHAB in a Docker container you need to make sure that UPnP discovery is available and network interfaces 
+
+## Generating the UID (`userId`)
+
+The binding requires a so called UID (`userId`) to pair with the receiver, which is generated by logging in with your Telekom credentials (Login Name + Password for the "Telekom Kundencenter").
+openHAB needs to access the Internet to perform that operation, which can be disabled afterwards.
+
+The binding initiates the pairing with the receiver once the userId has been obtained, this is an automated process.
```

Comment:
Is there really some "pairing" process happening here? If so, when? Everytime the thing handler is started? Or is it simply an "establishing connection" process?

Reply:
it creates some kind of logical connection between the receiver and the applcation. The MR sends the event updates to the applcation. This get renewed every time the binding starts and renewed periodically. It's a 3-stage process.